formatting.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+# Run it before committing to verify whether all files 
+# are properly formatted so you won't fail early on review
+# Execute from repo root or, if using aa from base/head/startup-os
+
+RED=$(tput setaf 1)
+RESET=$(tput sgr0)
+
+npm install &>/dev/null
+
+bazel run //tools/formatter -- \
+					--path $(pwd) \
+					--java --python --proto --cpp --build \
+					--ignore_directories $(find $(pwd) -name node_modules -type d | paste -s -d , -) \
+					&>/dev/null
+if [[ ! -z "$CIRCLECI" && ! -z $(git status -s) ]]; then
```

Comment:
`git status -s` means we're checking modified unstaged files?
What if I'm currently working and I have some modified files?
Also, if I run this twice, will it print the error on the second time?

Reply:
@oferb 
>git status -s means we're checking modified unstaged files?

Right

>What if I'm currently working and I have some modified files?

Then you want fall through `! -z "$CIRCLECI"` condition -- since you're *not* on CircleCI

>Also, if I run this twice, will it print the error on the second time?

It prints out an error only if both conditions are satisfied:
* we are on CircleCI
* working tree contains unstaged changes

When you run it locally it silently fixes everything.

Reply:
Great. Can you add that as a comment just before the `if`?